platforms/Cross/vm/sqMemoryFence.h

```diff
@@ -53,8 +53,18 @@
 #	endif
 # endif
 #elif defined(_MSC_VER)
-# include <intrin.h>
-# define sqLowLevelMFence() _ReadWriteBarrier() /* Note that's is deprecated, superseded by C++11 features */
+  /* # define sqLowLevelMFence() _asm { mfence } */
+  /* could use mfence here but */ \
+  /* See http://shipilev.net/blog/2014/on-the-fence-with-dependencies/ */ \
+# if defined(_M_IX86) || defined(_X86_)
+#  define sqLowLevelMFence() __asm { \
+  __asm lock add [esp], 0 \
+  }
+# elif defined(_WIN64) || defined(_M_X64) || defined(_M_AMD64)
+#  define sqLowLevelMFence() __asm { \
```

Comment:
Oh, you're right :(  <https://msdn.microsoft.com/en-us/library/wbk4z78b.aspx> <http://stackoverflow.com/a/1295495/1197440>. This has to go...

We might want `MemoryBarrier` <https://msdn.microsoft.com/en-us/library/windows/desktop/ms684208(v=vs.85).aspx>